spec/features/post_index_spec.rb

```diff
@@ -0,0 +1,51 @@
+require 'rails_helper'
+
+RSpec.describe 'Post index', type: :feature do
+  before :each do
+    @user = User.create(name: 'Ross', photo: 'https://picsum.photos/id/23/200', bio: 'Teacher from Mexico.', posts_counter: 0)
+    @post1 = Post.create(author_id: @user.id, title: 'Hello', text: 'This is my first post', comments_counter: 0, likes_counter: 0)
+    @post2 = Post.create(author_id: @user.id, title: 'World', text: 'This is my second post', comments_counter: 0, likes_counter: 0)
+    @post3 = Post.create(author_id: @user.id, title: 'Hi', text: 'This is my third post', comments_counter: 0, likes_counter: 0)
+    @post4 = Post.create(author_id: @user.id, title: 'Hi World', text: 'This is my fourth post', comments_counter: 0, likes_counter: 0)
+    @comment = Comment.create(post_id: @post1.id, author_id: @user.id, text: 'This is my first comment')
+    @like = Like.create(post_id: @post1.id, author_id: @user.id)
+    visit user_posts_path @user.id
+  end
+
+  it 'can see user profile picture' do
+    expect(page).to have_css("img[src*='https://picsum.photos/id/23/200']")
+  end
+
+  it 'can see the username' do
+    expect(page).to have_content('Ross')
+  end
+
+  it 'can see the number of posts user has written' do
+    expect(page).to have_content(@user.posts_counter)
+  end
+
+  it 'can see post title' do
+    expect(page).to have_content 'Hello'
```

Comment:
- [ ] Please make sure to include the below test for the posts index page as requested in the project requirements.

![image](https://github.com/Rhaegar121/Blog-app/assets/83803414/491a8276-3d00-46f3-be31-1b7208eee2a1)